docs/editor-neovim.qmd

```diff
@@ -0,0 +1,52 @@
+---
+title: "Neovim"
+editor:
+  markdown:
+    wrap: sentence
+    canonical: true
+---
+
+# Installing the air client
+
+First, install the Air client by following the instructions on the [Air CLI documentation](https://posit-dev.github.io/air/cli.html).
+
+# Configuration
+
+Air can be configured as a Language Server Protocol (LSP) client via [nvim-lspconfig](https://github.com/neovim/nvim-lspconfig) or as a formatter plugin via [conform.nvim](https://github.com/stevearc/conform.nvim).
+
+## LSP Configuration with nvim-lspconfig
+
+To configure Air's language server, add the following to your config:
+
+```lua
+require("lspconfig").air.setup({})
+```
+
```

Comment:
```suggestion

This is the recommended way to use Air. While the language server currently only supports formatting, it will gain more features in future.

```

Reply:
Any thoughts about layering Air as an LSP with `languageserver`?

Reply:
@lionel- do you mean having both Air and `languageserver` enabled, but using only Air for formatting?

Reply:
Want to add it to the config instruction?

Reply:
Maybe something like

If both `Air` and `languageserver` are installed, you can use the following configuration to rely solely on `Air`'s formatting capabilities:

```lua
require("lspconfig").r_language_server.setup({
    on_attach = function(client, _)
        client.server_capabilities.documentFormattingProvider = false
        client.server_capabilities.documentRangeFormattingProvider = false
    end,
})
```

Reply:
I'd say it's to disable `languageserver` formatting. In theory, someone can have other LSPs attached. Either that or using the `filter` in `vim.lsp.buf.format()` that you showed.

I think the bottom line is: make sure the user is aware they need to change something as both of these have formatting capabilities.